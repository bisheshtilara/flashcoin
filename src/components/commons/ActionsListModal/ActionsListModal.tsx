import React from "react";
import ModalContainer from "../ModalContainer";

export interface IActionsListModalProps {
  isOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
  title: string;
  handleAction?: any;
}

const ActionsListModal: React.FC<IActionsListModalProps> = ({
  isOpen,
  setIsOpen,
  title,
  handleAction,
}) => {
  const content =
    "Are you sure you want to delete the selected item? This action is permanent and cannot be undone.";
  return (
    <ModalContainer
      isOpen={isOpen}
      setIsOpen={setIsOpen}
      title={title}
      action="Delete"
      buttonColor="bg-red-400"
      children={<p>{content}</p>}
      handleAction={handleAction}
    />
  );
};

export default ActionsListModal;
