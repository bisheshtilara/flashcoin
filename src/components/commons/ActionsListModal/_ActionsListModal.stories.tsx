import { ComponentMeta, ComponentStory } from "@storybook/react";
import ActionsListModal, { IActionsListModalProps } from "./ActionsListModal";

export default {
  component: ActionsListModal,
} as ComponentMeta<typeof ActionsListModal>;


const defaultArgs: IActionsListModalProps = {
  isOpen: true,
  setIsOpen: () => alert("clicked"),
  title: "Delete",
  handleAction: () => alert("action executed"),
};

const Template: ComponentStory<typeof ActionsListModal> = (args: IActionsListModalProps) => (
  <ActionsListModal {...args} />
);

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs,
};

