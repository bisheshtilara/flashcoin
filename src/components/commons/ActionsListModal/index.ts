import ActionsListModal from "./ActionsListModal";
export default ActionsListModal;