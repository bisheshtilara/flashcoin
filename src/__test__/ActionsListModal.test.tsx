import { render, fireEvent, screen } from "@testing-library/react";
import ActionsListModal from "../components/commons/ActionsListModal";

describe("ActionsListModal", () => {
    const setIsOpen = jest.fn();
    it("should render correctly", () => {
        const { container } = render(<ActionsListModal isOpen setIsOpen={setIsOpen} title='Test'/>);
        expect(container).toMatchSnapshot();
    });
    
    it("should render a modal have title Create test", () => {
        const { container } = render(<ActionsListModal isOpen setIsOpen={() => setIsOpen} title='Create test'/>);
        // expect(screen.getAllByText("Create test")).toHaveLength(1);
        const cancelButton = screen.getAllByRole("button");
        expect(cancelButton).toHaveLength(2);
    });
});

