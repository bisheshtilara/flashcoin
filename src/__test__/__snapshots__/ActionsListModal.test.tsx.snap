// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsListModal should render correctly 1`] = `
<div>
  <div
    class="block"
  >
    <div
      class="relative z-10"
    >
      <div
        class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"
      />
    </div>
    <div
      class="fixed inset-0 z-20 overflow-y-auto"
    >
      <div
        class="flex min-h-full justify-center items-center"
      >
        <div
          class="bg-white  flex flex-col justify-between w-[50%] rounded-lg"
        >
          <div
            class="flex items-center space-x-2 font-semibold border-b p-5"
          >
            <span
              class="text-2xl font-semibold"
            >
              Test
            </span>
          </div>
          <div
            class="p-10"
          >
            <p>
              Are you sure you want to delete the selected item? This action is permanent and cannot be undone.
            </p>
          </div>
          <div
            class="flex border-t p-2 space-x-2 justify-end"
          >
            <button
              class="border p-2 w-fit rounded-lg bg-red-400 text-white  hover:-translate-y-2 duration-300"
            >
              Delete
            </button>
            <button
              class="border p-2 w-fit rounded-lg bg-slate-400 text-white  hover:-translate-y-2 duration-300"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
